buildkite/resource_pipeline_schedule.go

```diff
@@ -155,17 +176,37 @@ func (ps *pipelineSchedule) Create(ctx context.Context, req resource.CreateReque
 func (ps *pipelineSchedule) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
 	var state pipelineScheduleResourceModel
 
-	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	diagsState := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diagsState...)
 
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	log.Printf("Reading Pipeline schedule %s ...", state.Label.ValueString())
-	apiResponse, err := getPipelineSchedule(ctx,
-		ps.client.genqlient,
-		state.Id.ValueString(),
-	)
+	timeouts, diags := ps.client.timeouts.Read(ctx, DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var apiResponse *getPipelineScheduleResponse
+	err := retry.RetryContext(ctx, timeouts, func() *retry.RetryError {
+		var err error
+
+		log.Printf("Reading Pipeline schedule %s ...", state.Label.ValueString())
+		apiResponse, err = getPipelineSchedule(ctx,
+			ps.client.genqlient,
+			state.Id.ValueString(),
+		)
+
+		if err != nil {
+			if isRetryableError(err) {
+				return retry.RetryableError(err)
+			}
+			return retry.NonRetryableError(err)
+		}
+		return nil
+	})
```

Comment:
Need to handle the case where there is no schedule found anymore and remove it from state. not sure how that looks like in this retry way of doing things

Reply:
I have added the logic to remove from state if schedule is not found. I think there  is no need to manage a  retry for that use-case.